woocommerce/wc-admin
# Store Profiler and Product task - include Subscriptions

Fixes #7445

This PR adds the Subscriptions inclusion to the Store Profiler and Product task. 


**Acceptance Criteria:**

- Remove the Subscriptions paid extension option from the Product Types step and include the Subscriptions free option.
- If the user selects the Subscriptions free option, display the following message at the bottom of the step: 
`The following extensions will be added to your site for free: WooCommerce Payments. An account is required to use this feature`.
- Install WooCommerce Payments in the Product Types step, if the user proceeds with the Subscriptions free option selected
- If the user visits the "Add product" task with WCP installed, include Subscriptions as an option in the product template modal
- If the user selects the subscriptions template option, redirect him to `post-new.php?post_type=product&subscription_pointers=true`

### Accessibility




### Detailed test instructions:

- Check out this branch.
- Deactivate and delete `WooCommerce Payments` if you have it installed.
- Go to the 3rd step of the store profiler (`Product Types`).
- Verify `Subscriptions` is shown as free (without a price chip). Also, verify that the text 
```
The following extensions will be added to your site for free: WooCommerce Payments. An account is required to use this feature
```
is visible at the bottom when `WooCommerce Payments` is not installed.

![screenshot-one wordpress test-2021 09 30-14_12_58](https://user-images.githubusercontent.com/1314156/135506696-b7812f7e-437f-4d89-956a-b73248f70f6b.png)

- Press `Continue` and verify that the `WooCommerce Payments` plugin is installed and activated and it's not shown in the `Free features` list

![screenshot-one wordpress test-2021 09 30-14_32_20](https://user-images.githubusercontent.com/1314156/135506727-d8888f2b-3424-4cf5-a4bf-b67a14a198b6.png)

- Go back to the `Home` screen. Check that the task item `Add Subscriptions to my store` is not visible in the setup task list.

![screenshot-one wordpress test-2021 09 30-14_39_28](https://user-images.githubusercontent.com/1314156/135506770-91571f8f-2e2e-43a7-b092-b9e5fdf56df8.png)

- Press `Add my products` in the setup task list.
- Select `Start with a template`. Verify that the option `Subscription product` is visible in the popup

![screenshot-one wordpress test-2021 09 30-14_35_22](https://user-images.githubusercontent.com/1314156/135506748-0b7bdce5-b006-47f9-9289-03ed26e4950c.png)

- Select `Subscription product` and press `Ok`. You should have been redirected to `post-new.php?post_type=product&subscription_pointers=true`
